src/utilities/frame/hooks.tsx

```diff
@@ -0,0 +1,27 @@
+import {useContext} from 'react';
+import {useAppBridge} from '../app-bridge';
+import {FrameContext} from './context';
+
+export function useFrame() {
+  const frame = useContext(FrameContext);
+  const appBridge = useAppBridge();
+
+  if (!frame && !appBridge) {
+    throw new Error(
+      'Your application must be wrapped in an <Frame> component to use: Loading, Toast, or ContextualSaveBar. See https://polaris.shopify.com/components/structure/frame for implementation instructions.',
```

Comment:
I don't think this should mention the Loading/Toast/ContextualSaveBar components explicitly - this is a level above those components and we might add more in the future.

Something like:

```
 'No Frame context was provided. Your component must be wrapped in a <Frame> component, or be used within an embedded application by setting the apiKey and shopOrigin properties on <AppProvider>. See https://polaris.shopify.com/components/structure/frame for implementation instructions.',
```